quipucords/fingerprinter/task.py

```diff
@@ -348,6 +348,10 @@ def _process_details_report(self, manager_interrupt, details_report):
                             fingerprint_dict.get('name')
                         insights_host['fqdn'] = \
                             fingerprint_dict.get('name')
+                        for fact in CANONICAL_FACTS:
```

Comment:
I think the canonical facts expected by the host inventory are slightly different than the canonical facts that are defined in quipucords. For example, the ones expected by host inventory are `bios_uuid`, `ip_addresses`, `mac_addresses`, `insights_client_id`, `rhel_machine_id`, `subscription_manager_id`. 

We will also need to update any affected test cases & update the swagger to include the new fields (canonical facts). 

One other thing - is we need to check and see if the system profile is empty before adding it to the insights host.

Reply:
Additionally, I think in the api spec, `insights_client_id` (from quipucords) will have to be renamed to `insights_id` when we add it to the host. 

We should talk to @kholdaway too because in one document, the `vm_uuid` is defined as a canonical fact - but there is no field for it in the api spec (which means we are only ever adding it to the nested facts field).